File: 50_excercies_lerner_renuven/5_pig_latin.py
EN_LETTERS = [
    'a', 'e', 'i', 'o', 'u', 'y',
    'A', 'E', 'I', 'O', 'U', 'Y'
]

def convertWordToPigLatin(word: str):
    word = word.strip()
    if len(word) == 0:
        return ''

    firstLetter = word[0]
    if firstLetter in EN_LETTERS:
        wordInPigLatinFormat = word + 'way'
    else:
        wordInPigLatinFormat = word[1::] + firstLetter + 'ay'

    return wordInPigLatinFormat

wordsForTest = ['fuck', 'donal', 'GAlexandr_', 'apple', 'any', 'x', 'a']
for wordForTest in wordsForTest:
    result = convertWordToPigLatin(wordForTest)
    print('WORD FOR TEST: ' + wordForTest + ', RESULT: ' + result)

File: 50_excercies_lerner_renuven/5_pig_latin.php
<?php

const EN_LETTERS = [
    'a', 'e', 'i', 'o', 'u', 'y',
    'A', 'E', 'I', 'O', 'U', 'Y'
];

function convertWordToPigLatin(string $word): string
{
    $word = trim($word);
    if (empty($word)) {
        return '';
    }

    $firstLetter = $word[0];
    if (in_array($firstLetter, EN_LETTERS)) {
        $wordInPigLatinFormat = $word . 'way';
    } else {
        $wordInPigLatinFormat = substr_replace($word, '', 0, 1) . $firstLetter . 'ay';
    }

    return $wordInPigLatinFormat;
}


$wordsForTest = ['fuck', 'donal', 'GAlexandr_', 'apple', 'any', 'x', 'a'];
foreach ($wordsForTest as $wordForTest) {
    $result = convertWordToPigLatin($wordForTest);
    print_r("WORD FOR TEST: $wordForTest, RESULT: $result \n");
}